open-api/rest-catalog-open-api.yaml

```diff
@@ -1453,6 +1456,23 @@ components:
         type: string
       example: "sales"
 
+    data-access:
+      name: X-Iceberg-Data-Access
```

Comment:
This doesn't seem like a clear enough noun to me. How about `X-Supported-Access-Delegation`? I think it should be clear that these are access patterns or methods that are supported by the client. It's also for metadata access as well (manifests and manifest lists). I think having Iceberg in there is nice, but probably not necessary if it is getting too long.

Reply:
Updated to `X-Iceberg-Access-Delegation`.  I think it's important to keep `X-Iceberg` as the prefix so there isn't confusion around what the header relates to (pretty standard convention).